src/hid_linux.c

```diff
@@ -97,8 +98,14 @@ parse_uevent(const char *uevent, int *bus, int16_t *vendor_id,
 				*vendor_id = (int16_t)y;
 				*product_id = (int16_t)z;
 				ok = 0;
-				break;
 			}
+			continue;
+		}
+		if (strncmp(p, "HID_NAME=", 9) == 0 && hid_name != NULL) {
+			if (sscanf(p + 9, "%79[^\n]s", name) == 1 && *hid_name == NULL) {
```

Comment:
Any particular reason we're not just calling `strdup(p + 9)`, and skipping the `sscanf()` parsing?

Reply:
Yes, actually:
`cp` holds the complete rest of the string of `uevent`, which potentially includes all of the newlines and following lines. `sscanf` already properly deals with 0 terminator and newlines.

As sscanf was already part of the function, I was thinking, this would be fair game, as long as the buffer length and everything are set accordingly.

Alternative solutions could use `strtok` , `strsep` or searching for newline and then use `strncpy`.

Do you want me to change it?

Reply:
Hm, not sure I follow.  We're already using `strsep()` call used to iterate over each line of the `uevent`:

https://github.com/Yubico/libfido2/blob/0071ac0e8d738b728acd92e4f143834a0d2ac578/src/hid_linux.c#L94

This should already guarantee that `p` does not contain any `\n` and is be NUL-terminated.